### PR TITLE
Adds best_fans first stage refactor

### PR DESCRIPTION
**Pull request by colintalex**
Adds worst_fans stage_1 refactor with helper methods

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -204,61 +205,22 @@ def highest_scoring_home_team
...
-      percentage_differences[team_id] = (home_percentage - away_percentage).abs
-    end
-
-    team_with_largest_difference = percentage_differences.max_by { |team_id, percentage_diff| percentage_diff}
-
-    team = @team_collection.teams.find do |team|
-      team.team_id == team_with_largest_difference[0]
-    end
-    team.teamname
+    #biggest diff between home win % and away win %
```
Comment (mikez321) [inline lib/stat_tracker.rb:208]: Much fewer lines!  Still can probably shorten this some though.

Comment (aperezsantos) [inline lib/stat_tracker.rb:208]: Let's try to fix the indentation.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -150,6 +150,7 @@ def worst_defense
 
   def lowest_scoring_visitor
```
Comment (mikez321) [inline lib/stat_tracker.rb:151]: Let's put this in the to-do list

File: test/stat_tracker_test.rb [hunk 3]
```diff
@@ -310,7 +310,7 @@ def test_it_can_display_home_or_away_ties_by_team
   #
   # ######### it5 ###############
   #
-  def test_it_knows_team_info
+      def test_it_knows_team_info
```
Comment (aperezsantos) [inline test/stat_tracker_test.rb:313]: Let's fix indentation here too.

**Review by aperezsantos — APPROVED** [review]

**Review by mikez321 — APPROVED** [review]

**Pull request merged**